diff --git a/src/apprt/gtk/class/surface.zig b/src/apprt/gtk/class/surface.zig
index 616ec54f5..1d92b8608 100644
--- a/src/apprt/gtk/class/surface.zig
+++ b/src/apprt/gtk/class/surface.zig
@@ -1256,6 +1256,7 @@ pub const Surface = extern struct {
             "SNAP_USER_DATA",
             "SNAP_DATA",
             "SNAP_COMMON",
+            "BAMF_DESKTOP_FILE_HINT",
         };
 
         // Use an arena because everything in this function is temporary.
