diff --git a/src/apprt/gtk/class/surface.zig b/src/apprt/gtk/class/surface.zig
index c26d0c1ef..03ca6ba75 100644
--- a/src/apprt/gtk/class/surface.zig
+++ b/src/apprt/gtk/class/surface.zig
@@ -1240,6 +1240,9 @@ pub const Surface = extern struct {
             env.remove("GDK_PIXBUF_MODULEDIR");
             env.remove("GDK_PIXBUF_MODULE_FILE");
             env.remove("GTK_PATH");
+            env.remove("XDG_CONFIG_HOME");
+            env.remove("XDG_CACHE_HOME");
+            env.remove("XDG_DATA_HOME");
         }
 
         // This is a hack because it ties ourselves (optionally) to the
